Date/MomentFormatConverter.php

```diff
@@ -59,6 +57,42 @@ class MomentFormatConverter
      */
     public function convert($format)
     {
-        return strtr($format, self::$formatConvertRules);
+        $size = strlen($format);
+        $foundOne = false;
+        $output = '';
+        //process the format string letter by letter
+        for ($i = 0; $i < $size; $i++) {
+            //if finds a '
+            if ($format[$i] == '\'') {
+                //if the next character are T' forming 'T', send a T to the output
+                if ($format[$i+1] == 'T' && $format[$i+2] == '\'') {
+                    $output = $output . 'T';
+                    $i = $i+2;
+                } else {
+                    //if it's no a 'T' then send whatever is inside the '' to the output, but send it inside [] (useful for cases like the brazilian translation that uses a 'de' in the date)
+                    $output = $output . '[';
+                    $temp = explode('\'', substr($format, $i+1))[0];
+                    $output = $output . $temp;
+                    $output = $output . ']';
+                    $i = $i+strlen($temp)+1;
+                }
+            } else {
+                //if don't find a ' then search all the rules, and see if any on them matchs
+                $foundOne = false;
+                foreach (self::$formatConvertRules as $key => $value) {
+                    if (substr($format, $i, strlen($key)) == $key) {
+                        $output = $output . $value;
+                        $foundOne = true;
+                        $i = $i+strlen($key)-1;
+                        break;
+                    }
+                }
+                //if no rule is matched, then just add the character to the output
+                if ($foundOne == false) {
```

Comment:
`if (!$foundOne)` would be better. Don't compare booleans to booleans. Name them so that it makes a nice sentence when prepended with `if`.